src/nasa_provider.js

```diff
@@ -0,0 +1,56 @@
+import axios from "axios";
+
+const createRequest = (path, params) => {
+    let request = `${process.env.MARS_PHOTOS_URL}${path}`;
+
+    if (!params) {
+        params = {};
+    }
+    params.api_key = process.env.API_KEY;
+
+    request += '?';
+    for (const [key, value] of Object.entries(params)) {
+        request += `${key}=${value}&`
+    }
+
+    return request;
```

Comment:
I do enjoy this, but there are a few easier ways to deal with this:
* We can use the built in URL module to create our URLs with query params : https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams 
* Axios allows the query params to be passed in alongside the request:
  * I think we can event set up and axios instance with the api key that will then be added to all of our requests!
```
axios.get('...', { params: { key: value}})
```